File: ruby/modeling-lesson/lesson04/models/creditcard.rb
class Creditcard
  attr_accessor :id, :customer_id, :card_number, :expiration, :nominee

  def initialize(id, customer_id, card_number, expiration, nominee)
    @id = id
    @customer_id = customer_id
    @card_number = card_number
    @expiration = expiration
    @nominee = nominee
  end
end

File: golang/modeling-lesson/lesson04/model/creditcard.go
package model

import "time"

// クレジットカード
type Creditcard struct {
	ID         int       // ID
	CustomerID int       // 会員顧客 ID
	CardNumber string    // カード番号
	Expiration time.Time // 有効期限
	Nominee    string    // 名義
}

func NewCreditcard(id, customerID int, cardNumber string, expiration time.Time, nominee string) *Creditcard {
	return &Creditcard{
		ID:         id,
		CustomerID: customerID,
		CardNumber: cardNumber,
		Expiration: expiration,
		Nominee:    nominee,
	}
}
